packages/mui-material/src/styles/createPalette.js

```diff
@@ -310,7 +310,7 @@ export default function createPalette(palette) {
       // E.g., shift from Red 500 to Red 300 or Red 700.
       tonalOffset,
       // The light and dark mode object.
-      ...modes[mode],
+      ...getModeBasePalette[mode](),
```

Comment:
```suggestion
      ...getModeBasePalette[mode]?.(),
```
?

Reply:
At this point, the opposite seems simpler 😄

Reply:
It's simple enough, the types are just unsound. `mode` is `any` typed, typescript can't act as a safety net here. Is there an existing intention of converting our code to typescript?

Reply:
> Is there an existing intention of converting our code to typescript?

@Janpot Yes, I believe we are trying to progressively migrate everything #15984.